src/Ormolu/Printer/Meat/Declaration/Value.hs

```diff
@@ -1324,3 +1338,17 @@ getEnclosingAnns = do
   case e of
     Nothing -> return []
     Just e' -> getAnns (RealSrcSpan e')
+
+-- Put 'Nothing's between elements when there is more than one line
+-- between them.
+splitGroups :: [Located a] -> [Maybe (Located a)]
+splitGroups xs =
+  concatMap
+    ( \(prev, curr) ->
+        let prevEnd = fmap srcSpanEndLine . unSrcSpan . getLoc =<< prev
```

Comment:
I think it would be better if you used a custom one-off data type isomorphic to `Maybe` here.